modules/tinymce/src/core/test/ts/browser/delete/CefDeleteTest.ts

```diff
@@ -140,4 +141,84 @@ describe('browser.tinymce.core.delete.CefDeleteTest', () => {
+        TinySelections.setSelection(editor, [ 0, 0 ], 1, [], 2);
+        TinyContentActions.keystroke(editor, ks());
+        TinyAssertions.assertCursor(editor, [ 0, 0 ], 1);
+        TinyAssertions.assertContent(editor, '<p>a</p>');
+      });
+
+      it('TINY-8729: should delete selected content when cef block is at the start and at the end', () => {
+        const editor = hook.editor();
+        editor.setContent('<p contenteditable="false">CEF</p><p>abc</p><p contenteditable="false">CEF</p>');
+        // actual content: <p data-mce-caret="before"></p><p contenteditable="false">CEF</p><p>abc</p><p contenteditable="false">CEF</p>
+        TinySelections.setSelection(editor, [ 0 ], 0, [], 4);
+        TinyContentActions.keystroke(editor, ks());
+        TinyAssertions.assertContent(editor, '');
+      });
+    });
+  });
+
+  context('inline cef is at the start/end of the content and covered with the selection', () => {
+    Arr.each([ Keys.delete, Keys.backspace ], (ks) => {
+      it('TINY-8729: should delete selected content when inline cef is at the start', () => {
+        const editor = hook.editor();
+        editor.setContent('<p><span contenteditable="false">CEF</span>abc</p>');
+        // actual content: <p>&#xFEFF<span contenteditable="false">CEF</span>abc</p>
+        TinySelections.setSelection(editor, [ 0 ], 0, [ 0, 2 ], 2);
+        TinyContentActions.keystroke(editor, ks());
+        TinyAssertions.assertCursor(editor, [ 0 ], 0);
+        TinyAssertions.assertContent(editor, '<p>c</p>');
+      });
+
+      it('TINY-8729: should delete selected content when inline cef is at the end', () => {
+        const editor = hook.editor();
+        editor.setContent('<p>abc<span contenteditable="false">CEF</span></p>');
+        // actual content: <p>&#xFEFF<span contenteditable="false">CEF</span>abc</p>
+        TinySelections.setSelection(editor, [ 0, 0 ], 1, [ 0 ], 2);
+        TinyContentActions.keystroke(editor, ks());
+        TinyAssertions.assertCursor(editor, [ 0 ], 1);
+        TinyAssertions.assertContent(editor, '<p>a</p>');
+      });
+
+      it('TINY-8729: should delete selected content when inline cef is at the start and at the end', () => {
+        const editor = hook.editor();
+        editor.setContent('<p><span contenteditable="false">CEF</span>abc<span contenteditable="false">CEF</span></p>');
+        // actual content: <p>&#xFEFF<span contenteditable="false">CEF</span>abc<span contenteditable="false">CEF</span></p>
+        TinySelections.setSelection(editor, [ 0 ], 0, [ 0 ], 4);
+        TinyContentActions.keystroke(editor, ks());
+        TinyAssertions.assertCursor(editor, [ 0 ], 0);
+        TinyAssertions.assertContent(editor, '');
+      });
+    });
+  });
+
+  context('Cleaning up after rng.deleteContens call', () => {
+    Arr.each([ Keys.backspace, Keys.delete ], (ks) => {
+      it('TINY-8729: should clean up empty nodes and padd empty block with bogus br', () => {
```

Comment:
Can we please add one for a list here with an inline cef as well? As that's a somewhat more complicated case as I mentioned in my previous review, or will that never trigger this logic?

For example, something like:
```
<ul><li><span contenteditable="false">CEF</span></li></ul>
```
which should end up with something like:
```
<ul><li><br data-mce-bogus="1"></li></ul>
```

- [ ] Add list test case unless it's impossible for this logic to run into that.